policies/templates/gcp_sql_maintenance_window_v1.yaml

```diff
@@ -42,29 +42,56 @@ spec:
         # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
         # See the License for the specific language governing permissions and
         # limitations under the License.
-        #
+        # 
         
         package templates.gcp.GCPSQLMaintenanceWindowConstraintV1
         
         import data.validator.gcp.lib as lib
         
+        # A violation is generated only when the rule body evaluates to true.
         deny[{
         	"msg": message,
         	"details": metadata,
         }] {
+        	# by default any hour accepted
+        	default_hours := {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23}
+        	spec := lib.get_default(input.constraint, "spec", "")
+        	parameters := lib.get_default(spec, "parameters", "")
+        	maintenance_window_hours := lib.get_default(parameters, "hours", default_hours)
+        	desired_maintenance_window_hours := get_default_when_empty(maintenance_window_hours, default_hours)
+        	exempt_list := lib.get_default(parameters, "exemptions", [])
+        
         	# Verify that resource is Cloud SQL instance and is not a first gen
         	# Maintenance window is supported only on 2nd generation instances
         	asset := input.asset
         	asset.asset_type == "sqladmin.googleapis.com/Instance"
         	asset.resource.data.backendType != "FIRST_GEN"
         
-        	# get the maintenance window object
-        	maintenance_window := lib.get_default(asset.resource.data.settings, "maintenanceWindow", {})
+        	# Check if resource is in exempt list
+        	matches := {asset.name} & cast_set(exempt_list)
+        	count(matches) == 0
         
-        	# non compliant when not found
-        	maintenance_window == {}
+        	# get instance settings
+        	settings := lib.get_default(asset.resource.data, "settings", {})
```

Comment:
Maybe I am mistaken here, but would it be simpler to assign the settings value and test at the same time? Do you need to go further if settings is not in the resource data? If you use the "=" operator that would make it cleaner.